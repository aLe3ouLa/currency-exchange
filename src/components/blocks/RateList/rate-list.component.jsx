import React from "react";
import { connect } from "react-redux";

const RateList = ({ rates: { rates } }) => {
  const renderedRates = rates.rates;
  return (
    <div>
      <p>Base: {rates.base}</p>
      {Object.keys(renderedRates).map((key, index) => {
        return key && <p key={index}>{`${key} -- ${renderedRates[key]}`}</p>;
      })}
    </div>
  );
};

const mapStateToProps = (state) => ({
  rates: state.rates,
});

export default connect(mapStateToProps)(RateList);
