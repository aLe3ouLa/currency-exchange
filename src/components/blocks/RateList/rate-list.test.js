import React from "react";
import ReactDOM from "react-dom";
import RateList from "./rate-list.component";
import { Provider } from "react-redux";
import { cleanup, render } from "@testing-library/react";

import { ThemeProvider } from "styled-components";
import { theme, Defaults } from "../../../styles";

import configureStore from "redux-mock-store";
const mockStore = configureStore([]);

describe("Rates tests", () => {
  afterEach(cleanup);

  let store;
  beforeEach(() => {
    store = mockStore({
      rates: {
        rates: {
          base: "GBP",
          rates: {},
          date: new Date(),
        },
      },
      active: {
        active: {
          fromCurrency: "EUR",
          fromValue: 0,
          toCurrency: "GBP",
          toValue: 0,
        },
      },
      balance: {
        balance: {
          EUR: 12.01,
          GBP: 0.0,
          USD: 10.01,
          JPY: 0.0,
          lastExchange: {
            fromCurrency: "GBP",
            fromValue: 0,
            toCurrency: "EUR",
            toValue: 0,
            date: new Date(),
          },
        },
      },
      isLoading: false,
    });
  });

  it("renders the Rates component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <React.Fragment>
            <Defaults />
            <RateList />
          </React.Fragment>
        </ThemeProvider>
      </Provider>,
      div
    );
  });

  it("has a valid snapshot ", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <React.Fragment>
            <Defaults />
            <RateList />
          </React.Fragment>
        </ThemeProvider>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
