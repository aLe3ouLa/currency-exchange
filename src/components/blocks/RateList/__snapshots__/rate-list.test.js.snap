// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rates tests has a valid snapshot  1`] = `
<DocumentFragment>
  <div>
    <p>
      Base: GBP
    </p>
  </div>
</DocumentFragment>
`;
